### PR TITLE
feat: add snapshot to addressResolvers

### PR DESCRIPTION
**Pull request by wa0x6e**
Fix #262 

Lookup snapshot user's name from addresses

### 🛠️ Tests
 
Run 

```shell
curl --location 'http://localhost:3008' \
--header 'Content-Type: application/json' \
--data '{
    "method": "lookup_addresses",
    "params": [
        "0x91fd2c8d24767db4Ece7069AA27832ffaf8590f3", "0xeF8305E140ac520225DAf050e2f71d5fBcC543e7", "0x24F15402C6Bb870554489b2fd2049A85d75B982f", "0x76A6D08b82034b397E7e09dAe4377C18F132BbB8"
    ]
}'
```

It should return a list of addresses, with the username from snapshot for the first 3 addresses, and ENS name for the last one

### Other changes

- Add `address` and `name` to the cache clearing endpoint, so that calling `http://localhost:3008/clear/address/0x91fd2c8d24767db4Ece7069AA27832ffaf8590F3` will clear the address cache for the given address. This will be called by sequencer on user's name change, via https://github.com/snapshot-labs/snapshot-sequencer/pull/410

### CODE REVIEW

**Comment by wa0x6e** [comment]
mhhh. wait , test are failing

**Comment by wa0x6e** [comment]
Test passing now, just a temp thing

File: src/addressResolvers/index.ts [hunk 1]
```diff
@@ -12,7 +13,13 @@ import {
 import { Address, Handle } from '../utils';
 import { timeAddressResolverResponse as timeResponse } from '../helpers/metrics';
 
-const RESOLVERS = [ensResolver, unstoppableDomainResolver, lensResolver, starknetResolver];
+const RESOLVERS = [
+  snapshotResolver,
```
Comment (ChaituVR) [inline src/addressResolvers/index.ts:17]: What happens when the user edits his profile? 🤔 we will still have it in our cache right

Comment (wa0x6e) [reply src/addressResolvers/index.ts:17]: Good point … what to do

Comment (bonustrack) [reply src/addressResolvers/index.ts:17]: Ideally the sequencer would trigger stamp for clear cache of that specific user who updated his profile

Comment (wa0x6e) [reply src/addressResolvers/index.ts:17]: Cache clearing for `address` added. will make a PR on sequencer side to call it on user profile update

File: src/api.ts [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 import resolvers from './resolvers';
 import constants from './constants.json';
 import { rpcError, rpcSuccess } from './helpers/utils';
-import { lookupAddresses, resolveNames } from './addressResolvers';
+import { lookupAddresses, resolveNames, clearCache } from './addressResolvers';
 import lookupDomains from './lookupDomains';
 
 const router = express.Router();
-const TYPE_CONSTRAINTS = Object.keys(constants.resolvers).join('|');
+const TYPE_CONSTRAINTS = [...Object.keys(constants.resolvers), 'address', 'name'].join('|');
```
Comment (ChaituVR) [inline src/api.ts:12]: I don't remember where we use `name` resolver 🤔

**Review by ChaituVR — APPROVED** [review]
tAck

File: src/api.ts (hunk 2 above)
Comment (wa0x6e) [reply src/api.ts:12]: It's for resolveName, when passing a domain name (like chaitu.eth), and return an address.

It's there, but not also sure where we use it

**Pull request merged**